### PR TITLE
Doxygen updates + fixes for building

### PR DESCRIPTION
**Pull request by rem1776**
**Description**
Makes some adjustments to how autotools builds the docs to fix some issues, in addition to the documentation updates already added in #625 (had to make a second PR because i couldn't edit it directly)

Fixes #

**How Has This Been Tested?**
../configure --enable-docs && make -C docs
 
**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by underwoo** [comment]
@rem1776 there appear to be merge conflicts in the PR. Please resolve and push an update.

**Comment by rem1776** [comment]
this is ready for review, any merge conflicts were resolved with d80edbb and c903988

**Comment by thomas-robinson** [comment]
I think we need #331 before we can merge this

**Review by wrongkindofdoctor — DISMISSED** [review]

**Review by wrongkindofdoctor — DISMISSED** [review]

**Review by bensonr — APPROVED** [review]

**Comment by bensonr** [comment]
@rem1776 - were any of the mpp/ source files updated from the current HTML style to doxygen?  If not, is doxygen properly configured to parse the HTML when run?

**Comment by rem1776** [comment]
@bensonr It looks like Colin didn't update the mpp files so they're still in html. 

It doesn't seem like doxygen can parse the html as is, so I think in order to get the mpp documentation included it'll need to be updated so the documentation is within doxygen comment blocks.

**Comment by bensonr** [comment]
@rem1776 - it looks like there is some support for parsing html embedded as
documentation  https://www.doxygen.nl/manual/htmlcmds.html.  There might be
a config option to parse the inline html in mpp.

**Comment by underwoo** [comment]
From testing I did years ago, the HTML documentation is parsable by Doxygen, but the comment tags (`!`) need to be modified to what doxygen is expecting (`![<>!]`).  

It is also important to know that Doxygen **will not** parse comments within the body of a subroutine.  It only parses comments before the subroutine and at the head (right after the beginning of the routine and the input parameters).

**Review by thomas-robinson — CHANGES_REQUESTED** [review]
Please address the 4 comments

File: diag_manager/diag_axis.F90 [hunk 1]
```diff
@@ -1093,8 +1147,10 @@ END FUNCTION get_axis_num
   !     Returns index in axis set table corresponding to a given axis set name.
   !   </DESCRIPTION>
   !   <IN NAME="set_name" TYPE="CHARACTER(len=*)">Set name.</IN>
+  !> @brief Returns index in axis set table corresponding to a given axis set name
+  !! @return Returns index in axis set table corresponding to a given axis set name
```
[the remark this answers is not in the document]
Comment (rem1776) [reply diag_manager/diag_axis.F90:1151]: Every function at least has a brief description so should i delete all of the old html style comments then as well?

[the remark this answers is not in the document]
Comment (thomas-robinson) [reply diag_manager/diag_axis.F90:1151]: If the comments are covered by doxygen, then there's no need for the double documentation.  It needs to be cleaned up eventually.

File: diag_manager/diag_output.F90 [hunk 2]
```diff
@@ -817,14 +829,28 @@ END SUBROUTINE write_axis_meta_data_fms2_io
...
+    CHARACTER(len=*), INTENT(in) :: name !< Field name
+    CHARACTER(len=*), INTENT(in) :: units !< Field units
+    CHARACTER(len=*), INTENT(in) :: long_name !< Field's long name
+    REAL, OPTIONAL, INTENT(in) :: RANGE(2) !< Valid range (min, max).  If min > max, the range will be ignored
+    REAL, OPTIONAL, INTENT(in) :: mval !< Missing value, must be within valid range
+    INTEGER, OPTIONAL, INTENT(in) :: pack !< Packing flag.  Only valid when range specified.  Valid values:
+                                          !! 1 = 64bit
+                                          !! 2 = 32bit
+                                          !! 4 = 16bit
+                                          !! 8 =  8bit
```
[the remark this answers is not in the document]
Comment (rem1776) [reply diag_manager/diag_output.F90]: Replaced this with a table in 48d7e50

File: diag_manager/diag_axis.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (rem1776) [reply diag_manager/diag_axis.F90]: I cleaned up the redundant html comments with 48d7e50

File: diag_manager/diag_axis.F90 [hunk 3]
```diff
@@ -723,9 +754,10 @@ END SUBROUTINE get_diag_axis_domain_name
   !     Return the length of the axis ID given.
   !   </DESCRIPTION>
   !   <IN NAME="id" TYPE="INTEGER">Axis ID</IN>
+  !> @brief Return the length of the axis.
+  !> @return Integer length
```
[the remark this answers is not in the document]
Comment (rem1776) [reply diag_manager/diag_axis.F90]: Fixed wording in 48d7e50

**Review by thomas-robinson — APPROVED** [review]

**Review by bensonr — APPROVED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]

**Pull request merged**